index.d.ts

```diff
@@ -93,7 +105,7 @@ declare class PCancelable<ValueType> extends Promise<ValueType> {
 			argument3: Agument3Type,
 			argument4: Agument4Type,
 			argument5: Agument5Type,
-			onCancel: OnCancelFunction
+			onCancel: PCancelable.OnCancelFunction
 		) => PromiseLike<ReturnType>
```

Comment:
Could https://devblogs.microsoft.com/typescript/announcing-typescript-3-4/#higher-order-type-inference-from-generic-functions help reduce the boilerplate here?

Reply:
I don't think so. The problem here is that the `onCancel` argument is passed as last argument to `userFn`. And TS only allows rest/spread on tuple types in the last position of the tuple. Thus this is incompatible with how tuple types work in TS, it's not an inference issue.